kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/filesystem/CSVFileReader.java

```diff
@@ -87,11 +101,17 @@ public void initialize() {
     Preconditions.checkState(state.equals(ReaderWriterState.NEW),
         "A reader may not be opened more than once - current state:%s", state);
 
-    try {
-      this.incoming =  fs.open(path);
-      this.size = fs.getFileStatus(path).getLen();
-    } catch (IOException ex) {
-      throw new DatasetReaderException("Cannot open path: " + path, ex);
+    if (incoming != null) {
+      this.size = 1; // avoid divide by 0 errors
+    } else {
+      Preconditions.checkNotNull(fs, "FileSystem cannot be null");
+      Preconditions.checkNotNull(path, "Path cannot be null");
+      try {
+        this.incoming = fs.open(path);
+        this.size = fs.getFileStatus(path).getLen();
```

Comment:
What if the file length is zero? Wouldn't you still get a divide-by-zero error later?

BTW this is a bug fix that is independent of the issue being fixed here so should be fixed separately - in a different commit, if not a separate JIRA.